sdk/eventhub/Azure.Messaging.EventHubs.Processor/src/EventProcessorClient.cs

```diff
@@ -596,16 +612,14 @@ public virtual async Task StartProcessingAsync(CancellationToken cancellationTok
         }
 
         /// <summary>
-        ///   Signals the <see cref="EventProcessorClient" /> to begin processing events.  Should this method be called while the processor
-        ///   is running, no action is taken.
+        ///   Signals the <see cref="EventProcessorClient" /> to stop processing events.  Should this method be called while the processor
+        ///   is not running, no action is taken.
         /// </summary>
         ///
-        /// <param name="cancellationToken">A <see cref="CancellationToken"/> instance to signal the request to cancel the start operation.  This won't affect the <see cref="EventProcessorClient" /> once it starts running.</param>
-        ///
-        /// <exception cref="EventHubsException">Occurs when this <see cref="EventProcessorClient" /> instance is already closed.</exception>
-        /// <exception cref="InvalidOperationException">Occurs when this method is invoked without <see cref="ProcessEventAsync" /> or <see cref="ProcessErrorAsync" /> set.</exception>
+        /// <param name="cancellationToken">A <see cref="CancellationToken"/> instance to signal the request to cancel the stop operation.  If the operation is successfully canceled, the <see cref="EventProcessorClient" /> will keep running.</param>
         ///
-        public virtual void StartProcessing(CancellationToken cancellationToken = default) => StartProcessingAsync(cancellationToken).GetAwaiter().GetResult();
+        public virtual async Task StopProcessingAsync(CancellationToken cancellationToken = default)
+            => await StopProcessingImplAsync(true, cancellationToken).ConfigureAwait(false);
```

Comment:
Any reason why we're awaiting this rather than just returning the Task directly?

Reply:
This is what we have in our guideline: https://github.com/davidfowl/AspNetCoreDiagnosticScenarios/blob/master/AsyncGuidance.md#prefer-asyncawait-over-directly-returning-task

Reply:
Totally agree with the principle, but is this in guidelines we actually link. I don't see it, for example, in https://docs.microsoft.com/en-us/dotnet/standard/design-guidelines/ or our own at  https://azure.github.io/azure-sdk/dotnet_introduction.html. We should probably make sure that David's guidelines are (probably preferrably) merged with official docs, or choose to adopt them ourselves.

Or am I missing a link to these somewhere?

Reply:
This link is mentioned in our Coding Guidelines in OneNote

Reply:
I opened https://github.com/Azure/azure-sdk-for-net/issues/10072 to track migration to publicly viewable and "follow-able" guidance.

Reply:
That's fair;  perhaps I'm naive but I can't claim to be familiar with the error scenarios that are alluded to in that guidance.  The general guidance that I've heard elsewhere and have been following is basically "prefer returning the task directly for performance reasons if you don't use the value and don't have a need to ensure this call is represented in the stack trace."    

That said, I certainly have no intent on arguing with David Fowler's advice.   I agree with Heath on visibility, it'd be helpful to have something public to rally around.  Thanks for opening the issue!